tools/dm/shard-merge-scenario.md

```diff
@@ -0,0 +1,273 @@
+---
+title: DM 分片合并场景
+summary: 学习分片合并场景下如何使用 DM 同步数据
+category: tools
+---
+
+# Data Migration 分片合并场景
+
+本文介绍如何在分片合并场景中使用 Data Migration (DM)。使用场景中，三个上游 MySQL 实例的分库和分表数据需要同步至下游 TiDB 集群。
+
+## 上游实例
+
+假设上游库结构如下：
+
+- 实例 1
+
+    | 库 | 表 |
+    |:------|:------|
+    | user  | information, log_north, log_bak |
+    | store_01 | sale_01, sale_02 |
+    | store_02 | sale_01, sale_02 |
+
+- 实例 2
+
+    | 库 | 表 |
+    |:------|:------|
+    | user  | information, log_east, log_bak |
+    | store_01 | sale_01, sale_02 |
+    | store_02 | sale_01, sale_02 |
+
+- 实例 3
+
+    | 库 | 表 |
+    |:------|:------|
+    | user  | information, log_south, log_bak |
+    | store_01 | sale_01, sale_02 |
+    | store_02 | sale_01, sale_02 |
+
+## 同步需求
+
+1. 合并三个上游实例中的 `user`.`information` 表至下游 TiDB 中的 `user`.`information` 表。
+2. 合并三个上游实例中的 `user`.`log_{north|south|east}` 表至下游TiDB中的 `user`.`log_{north|south|east}` 表。
+3. 合并三个上游实例中的 `store_{01|02}`.`sale_{01|02}` 表至下游TiDB中的 `store`.`sale` 表。
+4. 从三个上游实例中的 `user`.`log_{north|south|east}` 表中过滤掉所有删除操作。
+5. 从三个上游实例中的 `user`.`information` 表中过滤掉所有删除操作。
+6. 从三个上游实例中的  `store_{01|02}`.`sale_{01|02}` 表中过滤掉所有删除操作。
+7. 过滤掉三个上游实例中的 `user`.`log_bak` 表。
+8. 因为 `store_{01|02}`.`sale_{01|02}` 表带有长整型的自增主键，将这些表合并至 TiDB 时会引发合并冲突。您需要修改相应自增主键以避免冲突。
+
+## 下游实例
+
+假设同步后下游库结构如下：
+
+| 库 | 表 |
+|:------|:------|
+| user | information, log_north, log_east, log_south|
+| store | sale |
+
+## 同步方案
+
+- 要满足同步需求 #1 和 #2, 配置 [表路由规则](/tools/dm/data-synchronization-features.md##table-routing) 如下：
+
+    ```yaml
+    routes:
+      ...
+      user-route-rule:
+        schema-pattern: "user"
+        target-schema: "user"
+    ```
+
+- 要满足同步要求 #3, 配置 [表路由规则](/tools/dm/data-synchronization-features.md##table-routing) 如下：
+
+    ```yaml
+    routes:
+      ...
+      store-route-rule:
+        schema-pattern: "store_*"
+        target-schema: "store"
+      sale-route-rule:
+        schema-pattern: "store_*"
+        table-pattern: "sale_*"
+        target-schema: "store"
+        target-table:  "sale"
+    ```
+
+- 要满足同步要求 #4 和 #5, 配置 [binlog 事件过滤规则](/tools/dm/data-synchronization-features.md#binlog-event-filtering) 如下：
+
+    ```yaml
+    filters:
+      ...
+      user-filter-rule:
+        schema-pattern: "user"
+        events: ["truncate table", "drop table", "delete"， "drop database"]
+        action: Ignore
+    ```
+
+    > **注意：** 同步要求 #4、#5 和 #7 中，所有对 `user` 库的删除操作都要被过滤，所以此处配置了库级别的过滤规则。但需要注意的是，`user` 库将来出现表的删除操作也都会被过滤。
```

Comment:
I think the original CN doc (<https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/use-cases/shard-merge.md>) for this line is better.